Confusing diagnostic when argument type in impl FnOnce() is not found

### Code

```Rust
type Foo = Bar;

fn check(f: impl FnOnce(Foo), val: Foo) {
    f(val);
}
```


### Current output

```
Compiling playground v0.0.1 (/playground)
error[E0412]: cannot find type `Bar` in this scope
 --> src/lib.rs:1:12
  |
1 | type Foo = Bar;
  |            ^^^ not found in this scope

error[E0618]: expected function, found `impl FnOnce(Foo)`
 --> src/lib.rs:4:5
  |
3 | fn check(f: impl FnOnce(Foo), val: Foo) {
  |          - `f` has type `impl FnOnce(Foo)`
4 |     f(val);
  |     ^-----
  |     |
  |     call expression requires function

Some errors have detailed explanations: E0412, E0618.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `playground` (lib) due to 2 previous errors
```


### Desired output

```
Compiling playground v0.0.1 (/playground)
error[E0412]: cannot find type `Bar` in this scope
 --> src/lib.rs:1:12
  |
1 | type Foo = Bar;
  |            ^^^ not found in this scope


Some errors have detailed explanations: E0412, E0618.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `playground` (lib) due to 2 previous errors
```


### Rationale and extra context

Saying that `impl FnOnce()` is not a function is confusing and distracts from the real issue. I think compiler should either report nothing on the line with the call or say something about unknown types in function signature.

### Other cases

```Rust
type Foo = Bar;

// Doesn’t exhibit this issue
fn check(f: impl FnOnce() -> Foo) {
    f();
}
```


### Anything else?

_No response_

--- a/compiler/rustc_hir_typeck/src/callee.rs
+++ b/compiler/rustc_hir_typeck/src/callee.rs
@@ -581,6 +581,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         callee_ty: Ty<'tcx>,
         arg_exprs: &'tcx [hir::Expr<'tcx>],
     ) -> ErrorGuaranteed {
+        // Callee probe fails when APIT references errors, so suppress those
+        // errors here.
+        if let Some((_, _, args)) = self.extract_callable_info(callee_ty)
+            && let Err(err) = args.error_reported()
+        {
+            return err;
+        }
+
         let mut unit_variant = None;
         if let hir::ExprKind::Path(qpath) = &callee_expr.kind
             && let Res::Def(def::DefKind::Ctor(kind, CtorKind::Const), _)

--- a/tests/ui/typeck/apit-with-error-type-in-sig.rs
+++ b/tests/ui/typeck/apit-with-error-type-in-sig.rs
@@ -1,0 +1,8 @@
+type Foo = Bar;
+//~^ ERROR cannot find type `Bar` in this scope
+
+fn check(f: impl FnOnce(Foo), val: Foo) {
+    f(val);
+}
+
+fn main() {}

--- a/tests/ui/typeck/apit-with-error-type-in-sig.stderr
+++ b/tests/ui/typeck/apit-with-error-type-in-sig.stderr
@@ -1,0 +1,9 @@
+error[E0412]: cannot find type `Bar` in this scope
+  --> $DIR/apit-with-error-type-in-sig.rs:1:12
+   |
+LL | type Foo = Bar;
+   |            ^^^ not found in this scope
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0412`.

